desc/objectives/_coils.py

```diff
@@ -597,3 +597,207 @@ def compute(self, params, constants=None):
         data = tree_flatten(data, is_leaf=lambda x: isinstance(x, dict))[0]
         out = jnp.concatenate([dat["torsion"] for dat in data])
         return out
+
+
+class ToroidalFlux(_Objective):
+    """Target the toroidal flux in an equilibrium from a magnetic field.
+
+    This objective is needed when performing stage-two coil optimization on
+    a vacuum equilibrium, to avoid the trivial solution of minimizing Bn
+    by making the coil currents zero. Instead, this objective ensures
+    the coils create the necessary toroidal flux for the equilibrium field.
+
+    Parameters
+    ----------
+    eq : Equilibrium, optional
+        Equilibrium that will be optimized to satisfy the Objective.
+    field : MagneticField
+        MagneticField object, the parameters of this will be optimized
+        to minimize the objective.
+    target : {float, ndarray}, optional
+        Target value(s) of the objective. Only used if bounds is None.
+        Defaults to eq.Psi. Must be broadcastable to Objective.dim_f.
+    bounds : tuple of {float, ndarray}, optional
+        Lower and upper bounds on the objective. Overrides target.
+        Both bounds must be broadcastable to to Objective.dim_f
+    weight : {float, ndarray}, optional
+        Weighting to apply to the Objective, relative to other Objectives.
+        Must be broadcastable to to Objective.dim_f
+    normalize : bool, optional
+        Whether to compute the error in physical units or non-dimensionalize.
+    normalize_target : bool
+        Whether target should be normalized before comparing to computed values.
+        if `normalize` is `True` and the target is in physical units, this should also
+        be set to True.
+    loss_function : {None, 'mean', 'min', 'max'}, optional
+        Loss function to apply to the objective values once computed. This function
+        is called on the raw compute value, before any shifting, scaling, or
+        normalization. Note: has no effect for this objective
+    deriv_mode : {"auto", "fwd", "rev"}
+        Specify how to compute jacobian matrix, either forward mode or reverse mode AD.
+        "auto" selects forward or reverse mode based on the size of the input and output
+        of the objective. Has no effect on self.grad or self.hess which always use
+        reverse mode and forward over reverse mode respectively.
+    source_grid : Grid, optional
```

Comment:
I think for consistency with the other field objectives this should be `field_grid`?